### ISSUE TITLE
letsencrypt - please support google domains dynamic dns

### ISSUE THREAD

**Issue by morhimi** [home-assistant/hassio-addons#1003]
Google have their own domains service, please support add their support for their dynamic dns feature (not related to the newly added Google Cloud DNS)

**Comment by klauern** [comment]
Here's a reference doc to explain how to run an updater manually.  Might be easy enough to add in, but I haven't touched anything HASS related, so I'm not sure:

https://medium.com/@jeremygale/how-to-set-up-a-free-dynamic-hostname-with-ssl-cert-using-google-domains-58929fdfbb7a

**Comment by klauern** [comment]
Actually, following that guide I linked above, I was able to do it a bit differently.  I used both the [DNS-O-Matic](https://github.com/marthoc/hassio-addons/tree/master/dnsomatic) add-on with the [Lets Encrypt](https://github.com/home-assistant/hassio-addons/tree/master/letsencrypt) addon to accomplish the two parts of associating the IP subdomain and issuing the certificate.

The config for the DNS-O-Matic add-on :

```json
{
  "hostname": "",
  "username": "klauern",
  "password": "mymagicpass",
  "update_interval": 3600
}
```
Leaving the `host: ""` as empty means that it will update all of the configured services under your DNS-O-Matic account.

![image](https://user-images.githubusercontent.com/4735/72745330-253a1f80-3b75-11ea-9909-d4b776f520a3.png)


I have the Let's Encrypt Add-On perform the check via the `http` challenge method:


```json
{
  "email": "me@myplace.com",
  "domains": [
    "my-domain"
  ],
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem",
  "challenge": "http",
  "dns": {}
}
```
